ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -687,13 +685,36 @@ public boolean getResults(List results) throws IOException {
         return false;
       }
 
-      if (ss == Utilities.StreamStatus.EOF) {
+      if (streamStatus == Utilities.StreamStatus.EOF) {
         driverContext.setResStream(context.getStream());
       }
     }
     return true;
   }
 
+  @SuppressWarnings("rawtypes")
+  private boolean getFetchingTableResults(List results) throws IOException {
+    // If result set serialization to thrift object is enabled, and if the destination table is indeed written using
+    // ThriftJDBCBinarySerDe, read one row from the output sequence file, since it is a blob of row batches.
+    if (driverContext.getFetchTask().getWork().isUsingThriftJDBCBinarySerDe()) {
+      maxRows = 1;
+    }
+    driverContext.getFetchTask().setMaxRows(maxRows);
+    return driverContext.getFetchTask().fetch(results);
+  }
+
+  private String getRow(ByteStream.Output bos, Utilities.StreamStatus streamStatus) {
+    String row;
```

Comment:
nit: Safer to make this `final` to ensure that there aren't multiple paths later if this code changes.

Reply:
final added.